pyqtgraph/functions.py

```diff
@@ -253,45 +253,26 @@ def mkColor(*args):
             c = args[0]
             if len(c) == 1:
                 try:
-                    return Colors[c]
+                    return QtGui.QColor(Colors[c])  # return copy
                 except KeyError:
                     raise ValueError('No color named "%s"' % c)
-            have_alpha = len(c) in [5, 9] and c[0] == '#'  # "#RGBA" and "#RRGGBBAA"
-            if not have_alpha:
-                # try parsing SVG named colors, including "#RGB" and "#RRGGBB".
-                # note that QColor.setNamedColor() treats a 9-char hex string as "#AARRGGBB".
-                qcol = QtGui.QColor()
-                qcol.setNamedColor(c)
+            # match hex color codes
+            match = re.match(r"#([0-9a-fA-F]{3,8})", c)
+            if match and len(match.group(1)) in [3, 4, 6, 8]:
+                c = match.group(1)
+                if len(c) < 6:
+                    # convert RGBA to RRGGBBAA
+                    c = "".join([x + x for x in c])
+                if len(c) < 8:
+                    # convert RRGGBB to RRGGBBAA
+                    c += "ff"
+                r, g, b, a = bytes.fromhex(c)
+            else:
+                # 'c' might be an SVG color keyword
+                qcol = QtGui.QColor(c)
                 if qcol.isValid():
                     return qcol
-                # on failure, fallback to pyqtgraph parsing
-                # this includes the deprecated case of non-#-prefixed hex strings
```

Comment:
Did we intend to throw a deprecation warning when the code falls through into the path for non #-prefixed hex codes?

Hex codes without leading '#' character have been out of the documentation for a while (two years?) now, but it seems that we've continued to just silently accept them.